actors/builtin/multisig/multisig_test.go

```diff
@@ -311,6 +310,135 @@ func TestApprove(t *testing.T) {
 	})
 }
 
+func TestCancel(t *testing.T) {
+	actor := msActorHarness{multisig.MultiSigActor{}, t}
+
+	receiver := newIDAddr(t, 100)
+	anne := newIDAddr(t, 101)
+	bob := newIDAddr(t, 102)
+	chuck := newIDAddr(t, 103)
+
+	richard := newIDAddr(t, 104)
+
+	const noUnlockDuration = int64(0)
+	const numApprovals = int64(2)
+	const txnID = int64(0)
+	const fakeMethod = abi.MethodNum(42)
+	var fakeParams = abi.MethodParams{1, 2, 3, 4, 5}
+	var sendValue = abi.NewTokenAmount(10)
+	var signers = []addr.Address{anne, bob}
+
+	builder := mock.NewBuilder(context.Background(), t, receiver).WithCaller(builtin.InitActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple propose and cancel", func(t *testing.T) {
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, numApprovals, noUnlockDuration, signers...)
+
+		// anne proposes a transaction
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.propose(rt, chuck, sendValue, fakeMethod, fakeParams)
+		rt.Verify()
```

Comment:
Very helpful comments, thank you.

Later, blocks of lines like this could be factored into a one-liner, `proposeAndVerify`.